internal/config/matcher.go

```diff
@@ -82,25 +94,24 @@ func (m *Matcher) ForRequest(tenant string, groups []string) *Matcher {
 		}
 	}
 
-	return m
+	// Copy the keys array for request-specific modifications
+	keysCopy := make([]string, len(m.Keys))
+	copy(keysCopy, m.Keys)
+
+	return newMatcher(keysCopy, m.MatcherOp, m.skipTenants, m.adminGroups)
```

Comment:
Wouldn't this be simpler? (Mostly I don't why we need the additional `newMatcher` func)

```suggestion
	return &Matcher{
		Keys:        slices.Clone(m.Keys),
		MatcherOp:   m.MatcherOp,
		skipTenants: maps.Clone(m.skipTenants),
		adminGroups: maps.Clone(m.adminGroups),
	}
```